### ISSUE TITLE
try/catch suggestion

### ISSUE THREAD

**Issue by truevis** [jeremytammik/DirectObjLoader#1]
I suggest adding try/catch around builder.AddFace

```
    try
    {
        builder.AddFace(new TessellatedFace(corners,
          ElementId.InvalidElementId));       
    }
    catch
    {
     // remember something went wrong here..
    }
```

**Comment by jeremytammik** [comment]
hi eric, 

thank you for the suggestion! 

why, though? 

the whole operation is wrapped in an exception handler anyway. do you have an example where some individual faces may fail, and you still want to retain the rest? 

if so, i'll add this right away. 

cheers

jeremy

**Comment by truevis** [comment]
Here is an OBJ file with errors in it: https://drive.google.com/file/d/0B_UPhfVzWDkxMWJEVnMtdWN0N28/view?usp=sharing

**Comment by jeremytammik** [comment]
what exact exception is thrown, please? you should never add a catch-all handler if you can avoid it.

**Comment by jeremytammik** [comment]
i do not see the sample OBJ file you refer to. where is it?

**Comment by jeremytammik** [comment]
i updated the code with a catch-all exception handler. i would like to narrow it down to the specific exception thrown, though.

**Comment by truevis** [comment]
I think this OBJ tries to make 2-vertex faces https://drive.google.com/file/d/0B_UPhfVzWDkxMWJEVnMtdWN0N28/view?usp=sharing

**Comment by jeremytammik** [comment]
thx, i'll take a look.

**Comment by jeremytammik** [comment]
nope, if you simply look at the OBJ in a text editor, you see that all faces have four vertices. maybe some of the four vertices list the same one or two vertices twice? i'll test... you should be able to debug this also, man  :-)

**Comment by truevis** [comment]
Even MeshLab complains about that OBJ:

**Comment by jeremytammik** [comment]
hi eric, thank you for the high_ball_glass.obj sample OBJ. i added the exception handler you suggest, and also added a result log in the debug console window reporting the number of faces that succeeded and failed on each shape.

**Comment by jeremytammik** [comment]
closed with release 2015.0.0.17.
